Packages/nadena.dev.modular-avatar/Runtime/ModularAvatarObjectRemover.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+
+namespace nadena.dev.modular_avatar.core
+{
+    [ExecuteInEditMode]
+    [AddComponentMenu("Modular Avatar/MA Object Remover")]
+    public class ModularAvatarObjectRemover : AvatarTagComponent
+    {
+        public bool hideInHierarchy = true;
+        public bool keepDisabled = true;
+        public bool removePrefabComponents;
+        public List<GameObject> objectsToRemove = new List<GameObject>();
+        private List<GameObject> _lastObjectsToRemove;
+
+        private void OnEnable()
+        {
+            objectsToRemove?.RemoveAll(x => !x);
+            OnListChange();
+        }
+
+        private new void OnDestroy()
+        {
+            if (Application.isPlaying) return;
+            hideInHierarchy = false;
+            removePrefabComponents = false;
+            Unhide();
+            RestorePrefabComponents();
+            objectsToRemove.Clear();
+            _lastObjectsToRemove = null;
+        }
+
+        public void OnListChange()
```

Comment:
Should we trigger a check for changes from OnValidate as well? (again, to try to catch changes made from editor scripts/prefab override applications/reverts/etc)

Reply:
Can't be really triggered fro OnValidate, unless with delayCall.
But at this point I think its all out of scope of this idea, whole idea is to have some easy way of getting rid of parts of prefab you dont want to use without need of breaking the prefab and ability to restore it in case you change mind. Allowing for quicker creation, better optimisation and cleaner project, so I don't see any use cases for such situations